tests/sys_irq_handler/main.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2019 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Interrupt handler test application
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ *
+ * This application demonstrates how to use the interrupt handler thread
+ * module `irq_handler`. For that purpose the main thread simulates interrupts
+ * from 2 interrupt sources with different priorities with a period
+ * of a half second. One interrupt source triggeres a second interrupt before
+ * the interrupts can be handled by the interrupt handler.
+ *
+ * To be able to simulate the interrupts by the main thread, the interrupt
+ * handler thread has to have a lower priority as the main thread for this
+ * example. Otherwise, the interrupts would be handled immediatly when they
+ * are generated by the main thread. Therefore the interrupt handler thread
+ * priority is set to THREAD_PRIORITY_MAIN+1 for this example.
+ * @}
+ */
+
+#include <errno.h>
+#include <stdio.h>
+
+#include "irq_handler.h"
+#include "thread.h"
+#include "xtimer.h"
+
+static char some_stack[THREAD_STACKSIZE_MAIN];
+static int i = 0;
+
+/* preallocated and initialized interrupt event objects */
+static irq_event_t _int1_event = IRQ_EVENT_INIT;
+static irq_event_t _int2_event = IRQ_EVENT_INIT;
+
+/* interrupt service routine for a simulated interrupt from source 1 */
+void _int1_service (void *arg)
```

Comment:
`static`? (Some for other functions below that aren't used outside of this compilation unit.)

Reply:
Done